torch/csrc/inductor/aoti_torch/c/shim_deprecated.h

```diff
@@ -0,0 +1,151 @@
+#ifndef AOTI_TORCH_SHIM_DEPRECATED
+#define AOTI_TORCH_SHIM_DEPRECATED
+
+#include <stddef.h>
+#include <stdint.h>
+
+// WARNING: This header file contains to-be-deprecated APIs for aten ops that
+// will be removed in the future. Now shims for aten ops are generated by
+// torchgen and locate in torch/csrc/inductor/aoti_torch/generated
+
+#ifdef __GNUC__
+#define AOTI_TORCH_EXPORT __attribute__((__visibility__("default")))
+#else // !__GNUC__
+#ifdef _WIN32
+// PyTorch2 doesn't currently work on Windows. Exporting these APIs can lead
+// to symbol clashes at link time if libtorch is included in a DLL and binary
+// that depends on the DLL. As a short term fix, we don't export the symbols.
+// In the long term, this will need to be addressed when Windows is supported.
+#ifdef OVRSOURCE
+// Do not export AOTI on Windows for internal builds
+#define AOTI_TORCH_EXPORT
+#else /* OVRSOURCE */
+#ifdef EXPORT_AOTI_FUNCTIONS
+#define AOTI_TORCH_EXPORT __declspec(dllexport)
+#else
+#define AOTI_TORCH_EXPORT __declspec(dllimport)
+#endif
+#endif /* OVRSOURCE */
+#else // !_WIN32
+#define AOTI_TORCH_EXPORT
+#endif // _WIN32
+#endif // __GNUC__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct AtenTensorOpaque;
+using AtenTensorHandle = AtenTensorOpaque*;
+
+using AOTITorchError = int32_t;
+#define AOTI_TORCH_SUCCESS 0
+#define AOTI_TORCH_FAILURE 1
+
+AOTI_TORCH_EXPORT AOTITorchError aoti_torch__embedding_bag(
```

Comment:
Please also mark these functions as deprecated similar to here, https://github.com/pytorch/pytorch/blob/bc9091a524a1ebe4de16af4dd8f442db7d1cb138/c10/core/ScalarType.h#L378